### DISCUSSION TITLE
MovingCameraScene integration

### DISCUSSION THREAD

**Discussion by francescotaurone** [jeertmans/manim-slides#185] [Q&A] [answered]
Is there any way to move the camera, as with the manim class "MovingCameraScene"?

**Comment by jeertmans** [comment]
Hello! Do you mean mean using the aforementioned class, or moving the camera around with your mouse?

**Reply by francescotaurone** [reply]
I mean animating the camera like with the method camera.frame.animate, part of the aforementioned class, while using Slide.

**Reply by jeertmans** [reply]
Aah ok got it!
I could add a new class. Did you try sub classing it in the same way as I do with ThreeDSlide?
https://github.com/jeertmans/manim-slides/blob/b3fd1d209e00e323dfc12d4407fe04b90aafb412/manim_slides/slide.py#L361

**Reply by francescotaurone** [reply]
It does seem to work! I assumed there would have been conflicts with methods involving the camera. Thanks!

**Reply by jeertmans** [reply]
Nice! Would you mind sharing an example code?
I may add support for this class in the future

**Reply by francescotaurone** [reply, marked as answer]
Sure! Here is a simplified snippet:

```
from manim import *
from manim_slides import Slide
class SubclassExample(Slide, MovingCameraScene):
    def construct(self):
        eq1 = MathTex(r"x = 1")
        eq2 = MathTex(r"x = 2")
        self.play(Write(eq1))
        self.next_slide()
        self.play(TransformMatchingTex(eq1, eq2),
            self.camera.frame.animate.scale(0.5)
        )
        self.wait()
```

**Reply by jeertmans** [reply]
Thanks!
